rasterio/windows.py

```diff
@@ -195,6 +213,33 @@ def from_bounds(left, bottom, right, top, transform,
         return crop(window, height, width)
 
 
+def int_reshape(window, pixel_precision=3):
+    """Converts floating point value Windows to integer value Windows.
+
+    Parameters
+    ----------
+    window : Window
+        Input window with floating point values.
+    pixel_precision : int
+        Rounding precision in decimal places.
+    Returns
+    -------
+    Window
+        A new Window
+    """
+    if isinstance(window, tuple):
```

Comment:
@sgillies suggestion: when detecting input `tuple` normalize first to a `Window` then you can always apply the logic from your `else` block.  (reduces potential sources of error in redundant calculations).

Also - I found the docstring a little confusing at first: it implies that all parts of the window (including offsets) will be integers in returned `Window` - which is not the case.  Maybe make that a little more clear?